Add next-gen beta documentation (#1565)
slackapi/bolt-js: docs/_future/triggers.md

---
title: Triggers
order: 7
slug: triggers
lang: en
layout: tutorial
permalink: /future/triggers
---
# Triggers <span class="label-beta">BETA</span>

<div class="section-content">
Triggers are but one of the three building blocks that make up next-generation Slack apps. You will encounter all three as you walk the path of building your Slack app:

1. Functions are what define the actions of your app
2. Workflows are made up of Functions
3. Triggers are how Workflows are executed (⬅️ you are here)

> Since Triggers are what trigger your Workflows, you need to have a Workflow before you can create a Trigger. Acquaint yourself with the [documentation on Workflows](/bolt-js/future/workflows), then head back here. We'll wait!

Reviewer: ```suggestion
Since Triggers kick off Workflows, so you need to have a Workflow before you can create a Trigger. Acquaint yourself with the [documentation on Workflows](/bolt-js/future/workflows), then head back here. We'll wait!
```

With the knowledge of Workflows within your noggin, let's take a look at the different types of Triggers, and how you can implement them into your new app. 

> You will come to many forks in this metaphorical road that is Trigger implementation. There are no wrong choices; all roads lead to your own next-generation Slack app. 

Reviewer: I figure this is straight from the api site and I'd leave that stuff there - IMO we can cut this line since it feels not so much to the point.

```bash
slack trigger delete --trigger-id FtABC123
```
---

### Trigger types {#types}

There are four types of Triggers, each one having its own specific implementation. To learn more about each trigger, visit the linked pages below.

| Trigger Type                     | Use case                                                      |
|----------------------------------|---------------------------------------------------------------|
| [Link Triggers](https://api.slack.com/future/triggers/link)           | Invoke a Workflow from anywhere in Slack                      |
| [Scheduled Triggers](https://api.slack.com/future/triggers/scheduled) | Invoke a Workflow at specific time intervals                  |
| [Event Triggers](https://api.slack.com/future/triggers/event)         | Invoke a Workflow when a specific event happens in Slack      |
| [Webhook Triggers](https://api.slack.com/future/triggers/webhook)     | Invoke a Workflow when a specific URL receives a POST request |

---

### Manage access {#manage-access}

A newly created Trigger will only be accessible to others inside a workspace once its creator has granted access.

> To manage what user (or many users) have access to run your Triggers, you'll use the `access` command.

Reviewer: ```suggestion
To manage what user (or many users) have access to run your Triggers, you'll use the `slack trigger access` command.
```

| Trigger Type                     | Use case                                                      |
|----------------------------------|---------------------------------------------------------------|
| [Link Triggers](https://api.slack.com/future/triggers/link)           | Invoke a Workflow from anywhere in Slack                      |
| [Scheduled Triggers](https://api.slack.com/future/triggers/scheduled) | Invoke a Workflow at specific time intervals                  |
| [Event Triggers](https://api.slack.com/future/triggers/event)         | Invoke a Workflow when a specific event happens in Slack      |
| [Webhook Triggers](#https://api.slack.com/future/triggers/webhook)     | Invoke a Workflow when a specific URL receives a POST request |

---

### Manage access {#manage-access}

A newly created Trigger will only be accessible to others inside a workspace once its creator has granted access.

To manage what user (or many users) have access to run your Triggers, you'll use the `access` command.

To learn more about this, visit the guide [here](https://api.slack.com/future/triggers#manage-access).

---

> ### Conclusion {#conclusion}

Reviewer 2: note to self: maybe from here we can direct them to the deploy guide

Reviewer 2: (additional note - we'll add that in once the deploy guide is ready! resolving this comment)
